tests/testthat/test-partial-frame.R

```diff
@@ -18,9 +18,15 @@ test_that("order of variables comes from data", {
   expect_named(vec_type_common(df, pf), c("x", "y"))
 })
 
-test_that("partial variables added to end if not in data", {
+test_that("partial variables are forgotten if not in data", {
   pf <- partial_frame(y = 1)
   df <- data.frame(x = 1)
-  expect_named(vec_type_common(pf, df), c("x", "y"))
-  expect_named(vec_type_common(df, pf), c("x", "y"))
+  expect_named(vec_type_common(pf, df), "x")
+  expect_named(vec_type_common(df, pf), "x")
+})
+
+test_that("can assert partial frames", {
+  partial <- partial_frame(col_name = character(), measure = character())
+  expect_true(vec_is(data.frame(x = 1, col_name = 2), partial))
```

Comment:
The logic doesn't seem quite right here — I think this should assert that `df` contains at least the columns in the partial frame.